eval/eval.go

```diff
@@ -536,7 +532,7 @@ func (e *evalContext) evaluateInterpolate(x *expr, repr *interpolateExpr) *value
 		if i.value != nil {
 			pv := e.evaluatePropertyAccess(x, i.value.accessors)
 			s, unknown, secret := pv.toString()
-			v.unknown, v.secret = v.unknown || unknown, v.secret || secret
+			v.unknown, v.secret = v.containsUnknowns() || unknown, v.containsSecrets() || secret
```

Comment:
I think we need similar changes for `toString` and `toJSON`

Reply:
`v.combine` is updated to handle this now.